dotCMS/src/main/java/com/dotcms/rest/api/v1/experiment/ExperimentResource.java

```diff
@@ -0,0 +1,358 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import net.minidev.json.annotate.JsonIgnore;
+import org.apache.commons.lang.RandomStringUtils;
+
+@Path("/v1/experiment")
+public class ExperimentResource {
+
+    public static final String EXPERIMENT_COOKIE_NAME = "experiment";
+    public static final String LOOK_BACK_WINDOW_COOKIE_NAME = "lookBackWindowCookie";
+
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    public ExperimentSelected triggerExperiment(@Context final HttpServletRequest request,
+            @Context final HttpServletResponse response)
+            throws DotSecurityException, DotDataException {
+
+        final List<Experiment> runningExperiments = ExperimentFactory.getRunningExperiments();
+        final Experiment experiment = runningExperiments.isEmpty() ? null : pickOneExperiment(runningExperiments);
+
+        final ExperimentSelected experimentSelected = new ExperimentSelected();
+
+        if (UtilMethods.isSet(experiment) && checkRules(experiment, request, response)) {
+
+            final boolean shouldRunExperiment = shouldRunExperiment(experiment);
+            setCookies(response, experiment, shouldRunExperiment);
+
+            if (shouldRunExperiment) {
+                final ExperimentVariant experimentVariant = pickUpVarriant(experiment, response);
+                experimentSelected.variant = new VariantSelected(
+                        experimentVariant.getVariant().getName(), experiment.getVariantURL(experimentVariant));
+                experimentSelected.url = experiment.getURL();
+            } else {
+                experimentSelected.name = "NONE";
+            }
+        } else {
+            experimentSelected.name = "NONE";
+        }
+
+        return experimentSelected;
+    }
+
+    private ExperimentVariant pickUpVarriant(final Experiment experiment,
+            HttpServletResponse response) {
+        final int randomValue = (int) (Math.random() * 100);
+        double acum = 0;
+
+        for (final ExperimentVariant variant : experiment.getVariants().getAll()) {
+            acum += variant.getTrafficPercentage();
+
+            if (randomValue < acum) {
+                response.addCookie(
+                        new Cookie("variant", variant.getVariant().getName()));
+                return variant;
+            }
+        }
+
+        throw new RuntimeException();
```

Comment:
How is this exception handled by Jersey? shouldn't return something more accurate?